plugins/net_plugin/net_plugin.cpp

```diff
@@ -1176,8 +1176,8 @@ namespace eosio {
 
    connection::connection( const string& endpoint, const string& listen_address )
       : peer_addr( endpoint ),
-        strand( my_impl->thread_pool.get_executor() ),
-        socket( new tcp::socket( my_impl->thread_pool.get_executor() ) ),
+        strand( boost::asio::make_strand(my_impl->thread_pool.get_executor()) ),
+        socket( new tcp::socket( strand ) ),
```

Comment:
Now that the socket is always being created with a strand (both here and via the listener), that should mean that all the `bind_executor`s to the `strand` can be removed. Not sure you want to tackle that now or not (going in to 1.0 probably not, but going in to main maybe)

Reply:
I left it mainly because it was targeting 1.0. At this point, maybe it would be cleaner. In many places you would have `boost::asio::post(socket.get_executor(), [](){})` instead which is not really any difference I don't think.

Reply:
You would have to move the timers to using the strand too (afaict not a problem) but after that it does seem like all 5 `bind_executor`s could be removed. I am not seeing the need for `boost::asio::post(socket.get_executor(), [](){})` in those same spots. Anyways, it's not something that needs to be done now just a clean up I noticed since we're constructing these more idiomaticly now.

Reply:
Oh, you were talking specifically about the `bind_executor`. I was thinking of the `boost::asio::post(strand,`.